src/dsl/sc.rs

```diff
@@ -1,57 +1,109 @@
-use std::hash::Hash;
+use std::{hash::Hash, rc::Rc};
 
 use halo2_proofs::arithmetic::Field;
 
 use crate::{
     ast::Circuit,
     compiler::{
-        cell_manager::CellManager, step_selector::StepSelectorBuilder, Compiler, CompilerConfig,
+        self,
+        cell_manager::{CellManager, SingleRowCellManager},
+        step_selector::{SimpleStepSelectorBuilder, StepSelectorBuilder},
+        CompilationUnit, Compiler, CompilerConfig,
     },
     ir::{
         assigments::AssigmentGenerator,
         sc::{MappingContext, SuperCircuit},
     },
 };
 
+use super::{lb::LookupTableRegistry, CircuitContext};
+
 pub struct SuperCircuitContext<F, MappingArgs> {
-    sc: SuperCircuit<F, MappingArgs>,
+    super_circuit: SuperCircuit<F, MappingArgs>,
+    sub_circuit_phase1: Vec<CompilationUnit<F>>,
+    tables: LookupTableRegistry<F>,
```

Comment:
Looks like LookupTableRegistry is cloned from the super circuit to the circuit, and eventually to the step type. Does it mean adding a lookup table at the circuit level doesn't update it at the super circuit level?

If this is an issue, we could modify the `sub_circuit` function in dsl/sc.rs to add lookup tables from circuit lookuptableregistry to super circuit lookuptableregistry. Alternatively, we could simply delete the tables field from SuperCircuitContext if it's not needed.

Reply:
Because it is made of `Arc<Mutex` the clone works as a shared reference

Reply:
Ahh this is clever. Got it now.